cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -280,6 +280,66 @@ void generate_depth_remappings(std::map<int, std::pair<std::vector<int>, std::ve
   return total_pages;
 }
 
+/**
+ * @brief Helper function to convert an encoding bitmask to a readable string
+ *
+ * @param bitmask Bitmask of found unsupported encodings
+ * @returns Human readable string with unsupported encodings
+ */
+std::string encoding_bitmask_to_str(int8_t encoding_bitmask)
+{
+  std::string result;
+  if (encoding_bitmask & static_cast<int8_t>(unsupported_encoding_error::GROUP_VAR_INT)) {
+    result.append("GROUP_VAR_INT ");
+  }
+  if (encoding_bitmask & static_cast<int8_t>(unsupported_encoding_error::BIT_PACKED)) {
+    result.append("BIT_PACKED ");
+  }
+  if (encoding_bitmask & static_cast<int8_t>(unsupported_encoding_error::DELTA_LENGTH_BYTE_ARRAY)) {
+    result.append("DELTA_LENGTH_BYTE_ARRAY ");
+  }
+  if (encoding_bitmask & static_cast<int8_t>(unsupported_encoding_error::BYTE_STREAM_SPLIT)) {
+    result.append("BYTE_STREAM_SPLIT ");
+  }
+  if (!result.empty()) {
+    result.pop_back();  // Remove the last space.
+  }
+
+  return result;
+}
+
+/**
+ * @brief Create a readable string for the user that will list out all unsupported encodings found.
+ *
+ * @param pages List of page information
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @returns Human readable string with unsupported encodings
+ */
+std::string list_unsupported_encodings(const cudf::detail::hostdevice_vector<PageInfo>& pages,
+                                       rmm::cuda_stream_view stream)
+{
+  auto to_mask = [] __device__(const PageInfo& page) -> int8_t {
+    int8_t encoding_value =
+      page.encoding == Encoding::GROUP_VAR_INT
+        ? static_cast<int8_t>(unsupported_encoding_error::GROUP_VAR_INT)
+      : page.encoding == Encoding::BIT_PACKED
+        ? static_cast<int8_t>(unsupported_encoding_error::BIT_PACKED)
+      : page.encoding == Encoding::DELTA_LENGTH_BYTE_ARRAY
+        ? static_cast<int8_t>(unsupported_encoding_error::DELTA_LENGTH_BYTE_ARRAY)
+      : page.encoding == Encoding::BYTE_STREAM_SPLIT
+        ? static_cast<int8_t>(unsupported_encoding_error::BYTE_STREAM_SPLIT)
+        : 0x0;
+    return encoding_value;
+  };
```

Comment:
```suggestion
  auto to_mask = [] __device__(auto const& page) {
    return is_supported_encoding(page.encoding) ? 0U : encoding_to_mask(page.encoding);
  };
```
Then just change the template param for the `thrust::bit_or`.

Reply:
Sounds good! Tysm for your prompt replies :), will push changes shortly. I wasn't aware of the utility lying around so this is really helpful. Makes code a lot cleaner

Reply:
> I wasn't aware of the utility lying around

If I had a nickel for every time some kind soul pointed out a utility function I had missed... :moneybag: